onnxruntime/core/providers/cuda/tunable/cuda_tuning_context.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+
+#include "core/framework/tuning_context.h"
+#include "core/providers/cuda/cuda_execution_provider_info.h"
+
+namespace onnxruntime {
+
+class CUDAExecutionProvider;
+
+namespace cuda {
+namespace tunable {
+
+class CudaTuningContext : public ITuningContext {
```

Comment:
> `CudaTuningContext`

I noticed that `RocmTuningContext` is almost identical to `CudaTuningContext`. Can we move the `info_` and `manager_` fields to the base class as protected members? And also have the base class provide default (but overridable) implementations of `EnableTunableOp`, `DisableTunableOp` and `IsTunableOpEnabled`?

Reply:
`manager_` makes sense to be made into base class, but `info_` is not,  `TunableOpInfo` should be read as `cuda::TunableOpInfo` and `rocm::TunableOpInfo` and they are different at type level. The options in it might also diverge in the future.